File: app/src/main/java/com/metcalfe/ContributorDetailActivity.java
package alexmetcalfe.metcalfe;

import android.app.Activity;
import android.app.FragmentTransaction;
import android.os.Bundle;
import android.view.MenuItem;

import alexmetcalfe.j2objc_customer.R;

/**
 * Created by alex on 12/07/15.
 */
public class ContributorDetailActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_detail);

        getActionBar().setDisplayHomeAsUpEnabled(true);

        FragmentTransaction ft = getFragmentManager().beginTransaction();
        ft.add(R.id.content, ContributorDetailFragment.newInstance(getIntent().getIntExtra("contributions", 0))).commit();
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            finish();
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/kotlin/com/metcalfe/ContributorDetailActivity.kt
package com.metcalfe

import android.app.Activity
import android.os.Bundle
import android.view.MenuItem

/**
 * Created by alex on 12/07/15.
 */
class ContributorDetailActivity : Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_detail)

        actionBar!!.setDisplayHomeAsUpEnabled(true)

        val ft = fragmentManager.beginTransaction()
        ft.add(R.id.content, ContributorDetailFragment.newInstance(intent.getIntExtra("contributions", 0))).commit()
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            finish()
        }
        return super.onOptionsItemSelected(item)
    }
}
